src/main.cpp

```diff
@@ -1,31 +1,37 @@
 #include <stdio.h>
-#include <pico/stdio.h>
-#include <pico/multicore.h>
-#include <pico/flash.h>
+#include "pico/stdio.h"
+#include "pico/multicore.h"
+#include "pico/flash.h"
 // TinyUSB
 #include "bsp/board.h"
 #include "tusb.h"
 
-#include "pico_asha.h"
+#include "asha_unique_id.hpp"
+#include "asha_audio.hpp"
 
-#include "asha_audio.h"
+namespace asha
+{
 
-void asha_main();
-void usb_main(void);
+AudioBuffer audio_buff;
 
-#ifdef ASHA_PERF_TEST
-void perf_main(void);
-#endif
+patomic_bool pcm_streaming;
+patomic_bool encode_audio;
 
-struct asha_audio asha_shared;
-#ifdef ASHA_USB_SERIAL
-char pico_uid[2 * PICO_UNIQUE_BOARD_ID_SIZE_BYTES + 1 + 4];
-#else
-char pico_uid[2 * PICO_UNIQUE_BOARD_ID_SIZE_BYTES + 1];
-#endif
+char pico_uid[pico_uid_size];
+
+// extern "C" required by multicore_launch_core1
+extern "C" void bt_main();
 
-int main()
+void usb_main();
+
+extern "C" int main()
 {
+    // Init global shared variables
+    patom::PseudoAtomicInit();
+    audio_buff.init();
+    pcm_streaming = false;
+    encode_audio = false;
+
     // Get serial
     pico_get_unique_board_id_string(pico_uid, sizeof pico_uid);
 #ifdef ASHA_USB_SERIAL
```

Comment:
The `uid_suffix` is causing a compile error for me, because it doesn't have room for the null.

You can either do
```
const char uid_suffix[4] = {'-','C','D','C'};
```
or
```
const char uid_suffix[] = "-CDC";
memcpy(pico_uid + (sizeof(pico_uid) - sizeof(uid_suffix) - 2), uid_suffix, sizeof(uid_suffix)-1);
```

Reply:
Ah, oops. I forgot that switching to C++ changes this.

What I was doing is legal in C.

Reply:
FIxed. Thanks for pointing this out.